tools/jdk/java_toolchain_default.bzl

```diff
@@ -29,44 +36,49 @@ JDK9_JVM_OPTS = [
     "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
     "--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
 
-    # override the javac in the JDK.
-    "--patch-module=java.compiler=$(location :java_compiler_jar)",
-    "--patch-module=jdk.compiler=$(location :jdk_compiler_jar)",
-
     # quiet warnings from com.google.protobuf.UnsafeUtil,
     # see: https://github.com/google/protobuf/issues/3781
     # and: https://github.com/bazelbuild/bazel/issues/5599
     "--add-opens=java.base/java.nio=ALL-UNNAMED",
     "--add-opens=java.base/java.lang=ALL-UNNAMED",
 ]
 
-DEFAULT_JAVACOPTS = [
-    "-XDskipDuplicateBridges=true",
-    "-g",
-    "-parameters",
+# java_toolchain parameters without specifying javac, java.compiler,
+# and jdk.compiler module
+BASE_TOOLCHAIN_CONFIGURATION = dict(
```

Comment:
Nit: Any reason to use `dict(foo = "bar")` syntax over `{foo = "bar"}`?

Reply:
The second one was actually `{"foo": "bar"}`. I find the first one nicer: it is closer to actually calling rules with parameters and doesn't use so many strings.